components/builder-depot/src/backend/s3.rs

```diff
@@ -128,13 +146,19 @@ impl S3Handler {
         Counter::UploadRequests.increment();
         let key = s3_key(ident, target)?;
         let file = File::open(hart)?;
-        let path_attr = &hart.clone().into_os_string().into_string().unwrap();
+
         info!("S3Handler::upload request started for s3_key: {}", key);
+
         let size = file.metadata().unwrap().len() as usize;
         if size < MINLIMIT {
-            self.single_upload(key, file, &path_attr)
+            self.single_upload(&key, file, hart)
+//                .map_err(|e| Error::PackageUpload)
+                .and_then(move |_| {
+                    self.object_exists(&key)
```

Comment:
I wonder if this check should be feature flagged so we can turn these checks on and off as we want... (pulling in the usage of the feature  crate into builder depot will also be helpful for other future work btw)